compiler/luci/service/src/Nodes/CircleStridedSlice.cpp

```diff
@@ -92,19 +93,75 @@ struct StridedSliceContext
     params.new_axis_mask = node->new_axis_mask();
     params.shrink_axis_mask = node->shrink_axis_mask();
 
+    auto begin_node = loco::must_cast<luci::CircleNode *>(node->begin());
+    auto end_node = loco::must_cast<luci::CircleNode *>(node->end());
+    auto strides_node = loco::must_cast<luci::CircleNode *>(node->strides());
+
+    LUCI_ASSERT(begin_node->rank() == 1, "Only support rank 1 for begin_node");
+    LUCI_ASSERT(end_node->rank() == 1, "Only support rank 1 for end_node");
+    LUCI_ASSERT(strides_node->rank() == 1, "Only support rank 1 for strides_node");
+
     input = loco::must_cast<luci::CircleNode *>(node->input());
-    begin = loco::must_cast<luci::CircleConst *>(node->begin());
-    end = loco::must_cast<luci::CircleConst *>(node->end());
-    strides = loco::must_cast<luci::CircleConst *>(node->strides());
+    begin = dynamic_cast<luci::CircleConst *>(node->begin());
+    end = dynamic_cast<luci::CircleConst *>(node->end());
+    strides = dynamic_cast<luci::CircleConst *>(node->strides());
+    dummy.dtype(loco::DataType::S32);
+    dummy.rank(1);
+    dummy.shape_status(luci::ShapeStatus::VALID);
 
     loco::TensorShape input_shape = circle_shape(input);
+
+    if (begin == nullptr)
+    {
+      begin = &dummy;
+      begin->dim(0).set(begin_node->dim(0).value());
+      int32_t unknown_range = begin_node->dim(0).known() ? begin_node->dim(0).value() : input_dims;
+
+      for (int32_t i = 0; i < unknown_range; ++i)
+      {
+        input_shape.dim(i).unset();
+      }
+    }
+    if (end == nullptr)
+    {
+      end = &dummy;
+      end->dim(0).set(end_node->dim(0).value());
+      int32_t unknown_range = end_node->dim(0).known() ? end_node->dim(0).value() : input_dims;
+
+      for (int32_t i = 0; i < unknown_range; ++i)
+      {
+        input_shape.dim(i).unset();
+      }
+    }
+    if (strides == nullptr)
+    {
+      strides = &dummy;
+      strides->dim(0).set(strides_node->dim(0).value());
+      int32_t unknown_range =
+        strides_node->dim(0).known() ? strides_node->dim(0).value() : input_dims;
+
+      for (int32_t i = 0; i < unknown_range; ++i)
+      {
+        input_shape.dim(i).unset();
+      }
+    }
+
+    LUCI_ASSERT(begin->dtype() == S32, "Only support S32 for begin_node");
+    LUCI_ASSERT(end->dtype() == S32, "Only support S32 for end_node");
+    LUCI_ASSERT(strides->dtype() == S32, "Only support S32 for strides_node");
+
     input_dims = input_shape.rank();
+
+    assert(begin->size<S32>() <= input_dims);
+    assert(end->size<S32>() <= input_dims);
+    assert(strides->size<S32>() <= input_dims);
   }
   StridedSliceParams params;
   luci::CircleNode *input = nullptr;
   luci::CircleConst *begin = nullptr;
   luci::CircleConst *end = nullptr;
   luci::CircleConst *strides = nullptr;
+  luci::CircleConst dummy;
```

Comment:
@qsunki,
These changes are not related to the adding copyright part!?
Could it be that it was added incorrectly?

Reply:
This code was added while I was reviewing it, so I checked this part late. 😭

Reply:
Oh... another commit that I was testing got included... How should I fix this?

Reply:
Post a PR to remove your unintended changes.